docs/tutorials/fourier_checking.ipynb

```diff
@@ -0,0 +1,1187 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "view-in-github"
+   },
+   "source": [
+    "<a href=\"https://colab.research.google.com/github/ybc1991/Cirq/blob/forrelation/docs/tutorials/fourier_checking.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
```

Comment:
Done. Thanks for the notebook guide, which is really helpful! 

Just one comment: what I felt is not clear is what to determine the notebook has external dependency or not. 
For example, depending on networkx and pandas does not count as having external dependency because it is in the request in cirq. Can I understand this external dependency means any packages excluding cirq + its dependency?

Reply:
Yeah, I should use a different word there. I mean external API calls. Which this notebook does not have, so we should be able to test it. Added https://github.com/quantumlib/Cirq/issues/3803 to clarify this in the guide